sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/channel/ManagedChannelFactory.java

```diff
@@ -70,13 +92,42 @@ public ManagedChannel forDescriptor(ApiServiceDescriptor apiServiceDescriptor) {
    */
   private static class Default extends ManagedChannelFactory {
     @Override
-    public ManagedChannel forDescriptor(ApiServiceDescriptor apiServiceDescriptor) {
+    public ManagedChannel forDescriptor(
```

Comment:
As this method requires more than just descriptor, shall we rename it to something like `createChannel` or `getChannel`?